### ISSUE TITLE
Require a permission to break a block

### ISSUE THREAD

**Issue by PiggiesGoSqueal** [PluginBugs/Issues-ItemsAdder#1145]
**Is your feature request related to a problem? Please describe.**

I think a lot of server owners would find it useful to have a way to allow only specific players to mine X ItemsAdder ore/block type without being limited to certain vanilla or ItemsAdder tools. 

Normally this could be resolved in ItemsAdder by allowing only certain pickaxe types to mine X ore. Then making those certain pickaxes only usable by players with Y permission. 
- However, this whitelist pickaxe feature only supports vanilla and ItemsAdder pickaxes- not other ones such as those from MMOItems. 
- Furthermore, that method provides a different gameplay experience. It means you need better pickaxes to mine better ores. Whereas in some cases it may be more desirable to allow players to be able to mine ores with any type of pickaxe- as long as they have the permission to do so. 
  * The discouragement of mining a high tier ore with a low tier pickaxe could instead be reduced speed it takes to mine the ore/block.

### TLDR:
To summarize, requiring a permission to break X ItemsAdder block would provide a different gameplay experience and provide more control over how their players can mine ores/blocks without it being limited to the tools they use.

------------------
**Describe the solution you'd like**
Allow us to directly set a specific permission required to break a block.

------------------
**Describe alternatives you've considered**
- I've considered using the ItemsAdder whitelist pickaxe option so anyone with a certain pickaxe type can mine the ore. And then I could restrict the usage of the pickaxe to only those with a certain permission. However, as explained above this provides a different gameplay experience (slight while it may seem- it can have a large overall impact on a server's game design). It also restricts the tool usage to only vanilla & ItemsAdder tools. In my experience I've found each custom item plugin to have its own strengths and weaknesses and it is best to combine certain plugins to get the best overall experience- but limiting it to only itemsAdder tools prevents that.

- I've considered using MMOItems' block generation feature instead. However, when using MMOItems and ItemsAdder alongside each other, in short, this is either extremely difficult to implement or it isn't possible to implement (at least in the way I'd like to do it). And the MMOItems devs will not provide support or guides on the subject. I spent most of yesterday testing this and eventually came to the conclusion it isn't possible. 

- I've considered using another plugin that restricts who can mine what blocks. However, this would require them to add ItemsAdder support which is unlikely. Plus I have not seen a plugin that does this for 1.17.1.

------------------
**Additional context**
N/A

**Comment by LoneDev6** [comment]
Thanks a lot for the patience, the new update implements break/place permissions for both blocks and furniture.
```yml
info:
  namespace: block_permissions
items:
  block_1:
    enabled: true
    display_name: block
    permission_suffix: test.block_1
    resource:
      material: PAPER
      generate: false
      model_path: item/emerald
    specific_properties:
      block:
        placed_model:
          type: REAL_NOTE
        cancel_drop: false
        permission_suffix: ## <--- here
          break: test.block_1
          place: test.block_1
  furniture_1:
    enabled: true
    display_name: furniture
    permission_suffix: test.furniture
    resource:
      material: PAPER
      generate: false
      model_path: item/emerald
    behaviours:
      furniture:
        permission_suffix: ## <--- here
          break: test.furniture_1
          place: test.furniture_1
```

**Comment by PiggiesGoSqueal** [comment]
> Thanks a lot for the patience, the new update implements break/place permissions for both blocks and furniture.
> 
> ```yaml
> info:
>   namespace: block_permissions
> items:
>   block_1:
>     enabled: true
>     display_name: block
>     permission_suffix: test.block_1
>     resource:
>       material: PAPER
>       generate: false
>       model_path: item/emerald
>     specific_properties:
>       block:
>         placed_model:
>           type: REAL_NOTE
>         cancel_drop: false
>         permission_suffix: ## <--- here
>           break: test.block_1
>           place: test.block_1
>   furniture_1:
>     enabled: true
>     display_name: furniture
>     permission_suffix: test.furniture
>     resource:
>       material: PAPER
>       generate: false
>       model_path: item/emerald
>     behaviours:
>       furniture:
>         permission_suffix: ## <--- here
>           break: test.furniture_1
>           place: test.furniture_1
> ```

Awesome! Thanks
